x-pack/plugins/cases/public/components/edit_connector/helpers.ts

```diff
@@ -5,23 +5,33 @@
  * 2.0.
  */
 
+import { ConnectorTypeFields } from '../../../common';
 import { CaseUserActions } from '../../containers/types';
+import { parseStringAsConnector } from '../../containers/utils';
 
-export const getConnectorFieldsFromUserActions = (id: string, userActions: CaseUserActions[]) => {
+export const getConnectorFieldsFromUserActions = (
+  id: string,
+  userActions: CaseUserActions[]
+): ConnectorTypeFields['fields'] => {
   try {
     for (const action of [...userActions].reverse()) {
       if (action.actionField.length === 1 && action.actionField[0] === 'connector') {
-        if (action.oldValue && action.newValue) {
```

Comment:
The backend currently always sends the `oldValue` and `newValue` during a connector update but we don't _really_ need rely on that behavior. If we find the id in either we can just return it when we find it.